### PR TITLE
Update overview with latest CLI

### PR DESCRIPTION
**Pull request by gsans**
*Issue #, if available:*
Some recent changes were not available in the docs.

*Description of changes:*
Added new intro and create api.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/cli/restapi/restapi.md [hunk 1]
```diff
@@ -3,35 +3,139 @@ title: Overview
...
+A REST API or HTTP endpoint will be composed by one or more paths. Eg: `/items`. Each path will use a Lambda function to handle HTTP requests and responses. Amplify CLI creates a single resource in Amazon API Gateway so you can handle all routes, HTTP Methods and paths, with a single Lambda function via a Lambda Proxy integration. HTTP proxy integrations forward all requests and responses directly through to your HTTP endpoint.
+
+Amplify CLI let's you choose either an existing Lambda function or create a new one. To kickstart your implementation, you can choose between the following templates: 
+- Serverless ExpressJS function
+- CRUD function for DynamoDB
+
+> Lambda templates use [serverless-express](https://github.com/awslabs/aws-serverless-express) and provide the building blocks to start your REST API development.
+
+Amplify CLI supports the following Lambda runtimes:
```
Comment (renebrandel) [inline docs/cli/restapi/restapi.md:16]: I'd compress this into a sentence and point to the functions docs for the full list. This way when we add a new runtime, this doc will still be up-to-date.

File: docs/cli/restapi/restapi.md [hunk 2]
```diff
@@ -3,35 +3,139 @@ title: Overview
...
+See a description of these user types below
+
+| User type | Description |
+|---|---|
+| Authenticated user | User needs to sign in to use the REST API |
+| Guest user | User doesn't need to sign in to use the REST API |
+
+For each user type you can further specify what actions it has access to.
+
+| User type  | Actions | Http Method | Authentication Provider  |
```
Comment (renebrandel) [inline docs/cli/restapi/restapi.md:36]: @kaustavghosh06 can you confirm all of this?

File: docs/cli/restapi/restapi.md [hunk 3]
```diff
@@ -3,35 +3,139 @@ title: Overview
...
+|---|---|---|---|
+| Authenticated user | create, read, update, delete | POST, GET, PUT, PATCH, DELETE | Amazon Cognito |
+| Guest user | create, read, update, delete | POST, GET, PUT, PATCH, DELETE | None |
+
+
+REST APIs have support for [multiple environments](/cli/teams/overview.md) (e.g. dev, qa, and prod). This means that you can easily isolate different versions of your REST API by using different environments:
```
Comment (renebrandel) [inline docs/cli/restapi/restapi.md:42]: I believe all links need to have ~/... format

File: docs/cli/restapi/restapi.md (hunk 1 above)
Comment (gsans) [reply docs/cli/restapi/restapi.md]: See changes.

File: docs/cli/restapi/restapi.md (hunk 3 above)
Comment (gsans) [reply docs/cli/restapi/restapi.md]: Fixed

File: docs/cli/restapi/restapi.md (hunk 2 above)
Comment (kaustavghosh06) [reply docs/cli/restapi/restapi.md:31]: We also have user pool "groups" support for authentication other than  Authenticated and Guest user support. Should probably mention that out here.

File: docs/cli/restapi/restapi.md [hunk 4]
```diff
@@ -3,35 +3,134 @@ title: Overview
...
+|---|---|
+| Authenticated user | User needs to sign in to use the REST API |
+| Guest user | User doesn't need to sign in to use the REST API |
+
+For each user type you can further specify what actions it has access to.
+
+| User type  | Actions | Http Method | Authentication Provider  |
+|---|---|---|---|
+| Authenticated user | create, read, update, delete | POST, GET, PUT, PATCH, DELETE | Amazon Cognito |
+| Guest user | create, read, update, delete | POST, GET, PUT, PATCH, DELETE | None |
```
Comment (kaustavghosh06) [inline docs/cli/restapi/restapi.md:34]: Cognito is the provider for Guest/Unauthenticated users as well.

Comment (gsans) [reply docs/cli/restapi/restapi.md]: Fixed

**Pull request merged**